### PR TITLE
Add BigTreeTech Mini 12864 V1.0

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

Add [BigTreeTech Mini 12864 V1.0](https://github.com/bigtreetech/BGTHREETECH-MINI-12864-V1.0), which is similar to the Fysetc Mini 12864 2.1 with NeoPixels.

### Requirements

BigTreeTech Mini 12864 V1.0

### Benefits

Users can properly define LCD.

### CODE REVIEW

File: Marlin/src/inc/Conditionals_LCD.h [hunk 1]
```diff
@@ -31,8 +31,8 @@
   #define MKS_MINI_12864
 #endif
 
-// MKS_MINI_12864_V3 is simply identical to FYSETC_MINI_12864_2_1
-#if ENABLED(MKS_MINI_12864_V3)
+// MKS_MINI_12864_V3 and BTT_MINI_12864_V1 are simply identical to FYSETC_MINI_12864_2_1
+#if EITHER(MKS_MINI_12864_V3, BTT_MINI_12864_V1)
   #define FYSETC_MINI_12864_2_1
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_LCD.h:36]: Some pins files specifically refer to the `MKS_MINI_12864_V3` but not to `FYSETC_MINI_12864_2_1`. It looks like these pins files should use `FYSETC_MINI_12864_2_1` in place of `MKS_MINI_12864_V3` so they may be compatible with all three aliases.

Comment (thisiskeithb) [reply Marlin/src/inc/Conditionals_LCD.h:36]: Agreed.

File: Marlin/src/inc/Conditionals_post.h [hunk 2]
```diff
@@ -373,7 +373,7 @@
 #elif EITHER(MKS_MINI_12864, ENDER2_STOCKDISPLAY)
   #define _LCD_CONTRAST_MIN  120
   #define _LCD_CONTRAST_INIT 195
-#elif ENABLED(MKS_MINI_12864_V3)
+#elif EITHER(MKS_MINI_12864_V3, BTT_MINI_12864_V1)
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_post.h:376]: `FYSETC_MINI_12864_2_1` is also not mentioned here.

Comment (thisiskeithb) [reply Marlin/src/inc/Conditionals_post.h:376]: I think it uses different LCD contrast settings or is inherited from a parent `FYSETC_MINI_12864_*` setting.

Comment (thisiskeithb) [reply Marlin/src/inc/Conditionals_post.h:376]: @tpruvot would know more about the Fysetc variants I think.

Comment (tpruvot) [reply Marlin/src/inc/Conditionals_post.h:376]:
```diff
-//#define FYSETC_MINI_12864_X_X  // Type C/D/E/F. No tunable RGB Backlight by default
+#define FYSETC_MINI_12864_X_X    // Type C/D/E/F. No tunable RGB Backlight by default
 //#define FYSETC_MINI_12864_1_2  // Type C/D/E/F. Simple RGB Backlight (always on)
 //#define FYSETC_MINI_12864_2_0  // Type A/B. Discreet RGB Backlight
 //#define FYSETC_MINI_12864_2_1  // Type A/B. Neopixel RGB Backlight
```

Comment (tpruvot) [reply Marlin/src/inc/Conditionals_post.h:376]: well i have the basic one, just a backlight pin basically, as test display on a skr mini, backlight color is selected on shop order... button might be RGB

**Comment by thisiskeithb** [comment]
Related to this, Marlin will initialize LCD contrast at 0 on first flash/boot if no LCD with contrast was used before, instead of the defined `_LCD_CONTRAST_INIT` setting.

You wouldn't want it to always use that build time setting in case the user sets their own through the LCD/g-code, but I was thinking of a way to fix that using EEPROM flags, but it may have to wait until we can save individual EEPROM settings for some "First Run" type flags.

Warnings could be added in the interim or documentation can be beefed up since I've seen the issue crop of for users quite a bit:

```cpp
//
// Warn users of common LCD brightness/contrast issues
//
#if HAS_LCD_BRIGHTNESS
  #warning "Enabled LCD has software-controlled brightness. Use M256 [B<brightness>] to set brightness if LCD is unreadable on first boot: https://marlinfw.org/docs/gcode/M256.html"
#endif

#if HAS_LCD_CONTRAST
  #warning "Enabled LCD has software-controlled contrast. Use M250 [C<contrast>] to set contrast if LCD is unreadable on first boot: https://marlinfw.org/docs/gcode/M250.html"
#endif
```

**Comment by tpruvot** [comment]
i dont think brightness is "really" required to read on these LCD displays, its like a calculator, but in the night, maybe :p

but CONTRAST is ... can be all black (pixels) or nothing

**Comment by thinkyhead** [comment]
> …Warnings could be added in the interim…

I don't think that is an appropriate use of `Warnings.cpp`. First boot / EEPROM reset should ensure that the contrast is at its most visible and the brightness is near its full brightness.

**Comment by thisiskeithb** [comment]
> First boot / EEPROM reset should ensure that the contrast is at its most visible and the brightness is near its full brightness.

Resetting EEPROM also works, but I'm not sure if there's a better/smarter way since users are still stuck with a blank LCD after flashing (and Marlin/board appears dead).

**Comment by thinkyhead** [comment]
How often will a user flash Marlin and have a valid EEPROM with the contrast / brightness set too low? After a flash, most of the time Marlin will find the EEPROM is invalid and revert to defaults.

**Comment by thisiskeithb** [comment]
> How often will a user flash Marlin and have a valid EEPROM with the contrast / brightness set too low? 

I run into it enough while testing hardware & helping out users that I thought about adding these warnings 😄 

> After a flash, most of the time Marlin will find the EEPROM is invalid and revert to defaults.

That only works with `EEPROM_AUTO_INIT`, but perhaps that could be enabled in Marlin by default now if the user has `EEPROM_SETTINGS` enabled?

**Pull request merged**